File: server/src/main/java/com/content/security/entity/Vaccinationrecord.java
package com.content.security.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "vaccinationrecord")
public class Vaccinationrecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "Vaccination_id", nullable = false)
    private Vaccination vaccination;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "vaccineoffering_id", nullable = false)
    private Vaccineoffering vaccineoffering;

    @Size(max = 45)
    @Column(name = "date", length = 45)
    private String date;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "vaccinationstatus_id", nullable = false)
    private Vaccinationstatus vaccinationstatus;

}

File: client/src/app/core/entity/vaccinationrecord.ts
import {VaccineOffering} from "./vaccineoffering";
import {VaccinationStatus} from "./vaccinationstatus";

export class VaccinationRecord{

  id: number;
  vaccineoffering?:VaccineOffering;
  date?:string;
  vaccinationstatus:VaccinationStatus;

  constructor(id: number,vaccineoffering:VaccineOffering,date:string,vaccinationstatus:VaccinationStatus) {
    this.id = id;
    this.vaccineoffering = vaccineoffering;
    this.vaccinationstatus = vaccinationstatus;
    this.date = date;
  }
}


